impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Gauge.js

```diff
@@ -0,0 +1,155 @@
+/*!
+ * Copyright 2024 Hitachi Vantara. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "pentaho/module!_",
+  "./Abstract",
+  "common-ui/echarts",
+  "pentaho/visual/util"
+], function(module, BaseView, echarts, util) {
+
+  "use strict";
+
+  return BaseView.extend(module.id, {
+
+    _buildSeries: function() {
+      var dataTable = this.model.data;
+      var range = dataTable.getColumnRange(dataTable.getNumberOfColumns() - 1);
+      var rangeLength = range.max.toString().length;
+      var div = Math.pow(10, rangeLength - 1);
+
+      return [
+        {
+          type: "gauge",
+          min: 0,
+          max: Math.ceil(range.max / div) * div,
+
+          pointer: {
+            width: 8,
+            length: "80%",
+            offsetCenter: [0, "8%"]
+          },
+          progress: {
+            show: true,
+            overlap: true,
+            roundCap: true
+          },
+          detail: {
+            width: "auto",
+            height: "auto",
+            legendHoverLink: true,
+            fontSize: 14,
+            color: "#fff",
+            backgroundColor: "inherit",
+            borderRadius: 3,
+            formatter: "{value}"
+          },
+          itemStyle: {
+            borderColor: "#fff",
+            borderWidth: 1
+          },
+          emphasis: {
+            label: {
+              fontSize: 20
+            }
+          }
+        }
+      ];
+    },
+
+    _buildData: function() {
+      var dataTable = this.model.data;
+
+      if(dataTable.originalCrossTable) {
+        dataTable = dataTable.originalCrossTable.toPlainTable({skipRowsWithAllNullMeasures: true});
+      }
+
+      var rowLength = dataTable.getNumberOfRows();
+      var colLength = dataTable.getNumberOfColumns();
+      var font = util.getDefaultFont(null, 12);
+
+      var records = [];
+
+      var maxWidth = this.model.width;
+      var midIdx = Math.floor(rowLength / 2); // Calculate the index of the middle element
+
+      var initX = 0; // Midpoint x-coordinate
+      var yPercentage = 100; // Constant y-coordinate in percentage
+      var separation = (maxWidth / (rowLength - 1)) * 0.8; // Dynamic separation between elements
```

Comment:
Perhaps, better named as `hSeparation`.
Also, what's the logic of the layout being calculated?

Reply:
The `hSeparation` value is calculated based on the maximum width available and number of elements to be fit, and then placed evenly separated within the available area leaving 10% of space on either side.

Reply:
Are these the positions for the  labels accompanying the values?
If so, shouldn't the layout take the values, and their x/y scales to determine placement?

Could you please provide a print-screen of the rendering of a Gauge chart, as here proposed, to help map the code to the actual visual elementws?

Reply:
The code from line 93 to 110 help determine the offset on the X-axis, the Y-axis values are being set in lines 117 and 122.
Below is the current behavior:
![image](https://github.com/pentaho/pentaho-platform-plugin-common-ui/assets/113503432/90cb496f-48ac-4e39-a18a-ca40a30eb330)

Reply:
That was helpful, thanks! Please refer to the follow-up comment: https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1751#discussion_r1496613412.